### PR TITLE
Expanding module tests to ensure added seq len functionality for Deepseek 671B model 

### PR DESCRIPTION
**Pull request by yalrawwashTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/33963

### Problem description
Module tests currently only cover limited range of seq len (ie. only one small value).

### What's changed
Need to expand this testing to ensure extensive sequence length testing to help debug hangs / accuracy issues.

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/20380950881) CI passes
- [x] [Deepseek tests](https://github.com/tenstorrent/tt-metal/actions/runs/20793545161) CI passes

### CODE REVIEW

**Review by yieldthought — APPROVED** [review]
If this doesn't cause CI failures we can merge

**Comment by yalrawwashTT** [comment]
We actually only wanted to test one seq length temporarily, and to also add the pytest.skip functionality in CI. Including multiple tests for 1x glx will take too long to run in CI, testing all seq lengths across all tests took me 5-6 hours.

File: models/demos/deepseek_v3/tt/generator.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def __init__(
             hf_config if hf_config is not None else AutoConfig.from_pretrained(self.model_path, trust_remote_code=True)
         )
         # self._ensure_max_seq_len(self.hf_config)
-        self.hf_config.max_seq_len = 1024  # TODO: Change this when needed?
+        self.hf_config.max_seq_len = 131072  # TODO: Change this when needed?
```
Comment (esmalTT) [inline models/demos/deepseek_v3/tt/generator.py]: Comment is confusing. Delete it or add more context.

File: models/demos/deepseek_v3/tests/test_moe_gate.py [hunk 2]
```diff
@@ -43,6 +54,10 @@ def test_forward_pass(
     set_deterministic_env,
 ):
     """Test forward pass against reference model."""
+
+    # Skip all prefill seq lengths except 128
+    if mode == "prefill" and seq_len != 128:
+        pytest.skip(f"Skipping prefill with seq_len={seq_len}, only keeping prefill 128")
```
Comment (esmalTT) [inline models/demos/deepseek_v3/tests/test_moe_gate.py]: Add more context to this message across all of these files so it is clear why we are skipping.

Something like "Skipping prefilling with seq_len={} since this would cause us to exceed our available CI workload time"

File: models/demos/deepseek_v3/tests/test_moe_experts.py [hunk 3]
```diff
@@ -75,7 +75,18 @@ def create_combined_state_dict(module_path: str, model_path: Path, state_dict: d
     "mode, seq_len",
     [
         ("decode", 128),
+        # Powers of 2 from 128 to 128K for prefill
+        ("prefill", 128),
```
Comment (esmalTT) [inline models/demos/deepseek_v3/tests/test_moe_experts.py]: Should we put these cases in a single place? These tuples are just data, we can store them in a variable somewhere.

File: models/demos/deepseek_v3/tests/test_model.py [hunk 4]
```diff
@@ -291,8 +291,18 @@ def run_test_forward_pass_dpmodel(
     "mode, seq_len, batch_size_per_row",
     [
         ("decode", 1, 32),
+        # Powers of 2 from 128 to 128K for prefill
```
Comment (esmalTT) [inline models/demos/deepseek_v3/tests/test_model.py]: Not sure we need this comment, it's just stating the obvious.

**Review by esmalTT — CHANGES_REQUESTED** [review]

**Review by esmalTT — APPROVED** [review]
Thanks for addressing my comments, nice work!

**Review by pprajapatiTT — APPROVED** [review]
